File: 228.汇总区间.cs
/*
 * @lc app=leetcode.cn id=228 lang=csharp
 *
 * [228] 汇总区间
 */

// @lc code=start
public partial class Solution
{
    public IList<string> SummaryRanges(int[] nums)
    {
        IList<string> result = new List<string>();
        int startNum = 0;
        int count = 0;
        for (int i = 0; i < nums.Length; i++)
        {
            int currentNum = nums[i];
            if (count == 0 || (startNum + count != currentNum))
            {
                if (count != 0)
                {
                    result.Add(GetString(startNum, count));
                }
                startNum = currentNum;
                count = 1;
            }
            else
            {
                count++;
            }
        }
        if (count != 0)
        {
            result.Add(GetString(startNum, count));
        }
        return result;
    }

    private string GetString(int startNum, int count)
    {
        if (count == 1)
        {
            return startNum.ToString();
        }
        else
        {
            return $"{startNum}->{startNum + count - 1}";
        }
    }
}
// @lc code=end

File: src/228.汇总区间.rs
/*
 * @lc app=leetcode.cn id=228 lang=rust
 *
 * [228] 汇总区间
 */

// @lc code=start
impl Solution {
    pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
        let mut result = Vec::new();

        let len = nums.len();
        let mut left = 0;
        while left < len {
            let mut right = left;
            loop {
                right += 1;
                if right >= len || nums[right] - nums[right - 1] != 1 {
                    break;
                }
            }

            if left == right - 1 {
                result.push(nums[left].to_string());
            } else {
                result.push(format!("{}->{}", nums[left], nums[right - 1]));
            }

            left = right;
        }

        return result;
    }
}
// @lc code=end